File: December 2024 Leetcode Solution/December-04.java
class Solution {
    public boolean canMakeSubsequence(String str1, String str2) {
        int j = 0, m = str2.length();
        for (int i = 0; i < str1.length() && j < m; ++i) {
            char c = str1.charAt(i);
            if (c == str2.charAt(j) || (c - 'a' + 1) % 26 + 'a' == str2.charAt(j)) {
                ++j;
            }
        }
        return j == m;
    }
}

File: December 2024 Leetcode Solution/December-04.cpp
class Solution {
public:
    bool canMakeSubsequence(string& str1, string& str2) {
        int j = 0, m = str2.size();
        for (char c : str1) {
            if (j < m && (c == str2[j] || (c + 1 - 'a') % 26 + 'a' == str2[j])) {
                if (++j == m) return true;
            }
        }
        return j == m;
    }
};
